### PR TITLE
Improve callChoreo by intergrating SecretResolveManager

### PR DESCRIPTION
**Pull request by ChanikaRuchini**
**Purpose**
>Adaptive script gives the API key alias and the key is resolving  inside the callChoreo function.

### CODE REVIEW

File: components/org.wso2.carbon.identity.conditional.auth.functions.choreo/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/choreo/CallChoreoFunctionImpl.java [hunk 1]
```diff
@@ -57,6 +57,8 @@
 
     private static final String URL_VARIABLE_NAME = "url";
     private static final String API_KEY_VARIABLE_NAME = "apiKey";
+    private static final String API_KEY_ALIAS_VARIABLE_NAME = "apiKeyAlias";
+    private static final String SECRET_TYPE = "adaptive";
```
Comment (senthalan) [inline components/org.wso2.carbon.identity.conditional.auth.functions.choreo/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/choreo/CallChoreoFunctionImpl.java:61]: Can we rethink this SECRET_TYPE name? `adaptive` seems to be a bit generic 
What about `CALL_CHOREO_ADAPTIVE_SCRIPT`

File: components/org.wso2.carbon.identity.conditional.auth.functions.choreo/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/choreo/CallChoreoFunctionImpl.java [hunk 2]
```diff
@@ -72,7 +74,13 @@ public void callChoreo(Map<String, String> connectionMetaData, Map<String, Objec
                 request.setHeader(ACCEPT, TYPE_APPLICATION_JSON);
                 request.setHeader(CONTENT_TYPE, TYPE_APPLICATION_JSON);
 
-                String apiKey = connectionMetaData.get(API_KEY_VARIABLE_NAME);
+                String apiKey;
+                if (!StringUtils.isEmpty(connectionMetaData.get(API_KEY_VARIABLE_NAME))) {
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.conditional.auth.functions.choreo/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/choreo/CallChoreoFunctionImpl.java:78]:
```suggestion
                if (StringUtils.isNotEmpty(connectionMetaData.get(API_KEY_VARIABLE_NAME))) {
```

File: components/org.wso2.carbon.identity.conditional.auth.functions.choreo/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/choreo/CallChoreoFunctionImpl.java (hunk 1 above)
Comment (ChanikaRuchini) [reply components/org.wso2.carbon.identity.conditional.auth.functions.choreo/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/choreo/CallChoreoFunctionImpl.java]: Resolved

File: components/org.wso2.carbon.identity.conditional.auth.functions.choreo/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/choreo/CallChoreoFunctionImpl.java (hunk 2 above)
Comment (ChanikaRuchini) [reply components/org.wso2.carbon.identity.conditional.auth.functions.choreo/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/choreo/CallChoreoFunctionImpl.java]: Resolved

**Review by senthalan — APPROVED** [review]

File: components/org.wso2.carbon.identity.conditional.auth.functions.choreo/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/choreo/CallChoreoFunctionImpl.java [hunk 3]
```diff
@@ -35,6 +36,9 @@
 import org.wso2.carbon.identity.application.authentication.framework.exception.FrameworkException;
 import org.wso2.carbon.identity.conditional.auth.functions.choreo.internal.ChoreoFunctionServiceHolder;
 import org.wso2.carbon.identity.conditional.auth.functions.common.utils.Constants;
+import org.wso2.carbon.identity.secret.mgt.core.exception.SecretManagementException;
+import org.wso2.carbon.identity.secret.mgt.core.model.ResolvedSecret;
+
 
 import java.io.IOException;
```
Comment (DinikaSen) [inline components/org.wso2.carbon.identity.conditional.auth.functions.choreo/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/choreo/CallChoreoFunctionImpl.java:43]: Shall we remove the extra blank line here?

File: components/org.wso2.carbon.identity.conditional.auth.functions.choreo/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/choreo/internal/ChoreoFunctionServiceHolder.java [hunk 4]
```diff
@@ -32,6 +33,8 @@
     private ServerConfigurationService serverConfigurationService;
     private KeyStore trustStore;
     private ClientManager clientManager;
+    private static SecretResolveManager secretConfigManager;
+
```
Comment (DinikaSen) [inline components/org.wso2.carbon.identity.conditional.auth.functions.choreo/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/choreo/internal/ChoreoFunctionServiceHolder.java:38]: Remove extra blank line

File: components/org.wso2.carbon.identity.conditional.auth.functions.choreo/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/choreo/CallChoreoFunctionImpl.java (hunk 3 above)
Comment (ChanikaRuchini) [reply components/org.wso2.carbon.identity.conditional.auth.functions.choreo/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/choreo/CallChoreoFunctionImpl.java:43]: Addressed with commit [97b8dc8](https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/85/commits/97b8dc822153f28e50e8f878d46d78b6d6dfd1c0)

File: components/org.wso2.carbon.identity.conditional.auth.functions.choreo/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/choreo/internal/ChoreoFunctionServiceHolder.java (hunk 4 above)
Comment (ChanikaRuchini) [reply components/org.wso2.carbon.identity.conditional.auth.functions.choreo/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/choreo/internal/ChoreoFunctionServiceHolder.java:38]: Addressed with commit [97b8dc8](https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/85/commits/97b8dc822153f28e50e8f878d46d78b6d6dfd1c0)

**Pull request merged**